ops/move_deps.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import subprocess
+import shutil
+from pathlib import Path
+
+MISSING_REGEX = re.compile(r'\n\t(.+)\ =>\ not\ found')
+FOUND_REGEX = re.compile(r'\n\t(.+)\ =>\ (.+)\ (\(0[xX][0-9a-fA-F]+\))')
+
+def ldd(path):
+    """Get output of ldd for given file"""
+    ldd_out = subprocess.run(
+        ['ldd', path], check=True, capture_output=True, text=True
+    )
+    return ldd_out.stdout
+
+
+def get_missing_deps(ldd_output):
+    """Return iterator of missing dependencies in ldd output"""
+    for match in MISSING_REGEX.finditer(ldd_output):
+        yield match.group(1)
+
+def path_contains(parent, child):
+    """Check if first path contains the child path"""
+    parent = os.path.abspath(parent)
+    child = os.path.abspath(child)
+    return os.path.commonpath([parent]) == os.path.commonpath([parent, child])
+
+def get_deps_map(ldd_output, required_dir=None):
+    """Return dictionary mapping library names to paths"""
+    deps_map = {}
+    for match in FOUND_REGEX.finditer(ldd_output):
+        if required_dir is None or path_contains(required_dir, match.group(2)):
+            deps_map[match.group(1)] = match.group(2)
+    return deps_map
+
+
+def move_dependencies():
+    fil_lib = os.getenv('FIL_LIB', 'libtriton_fil.so')
+    lib_dir = os.getenv('LIB_DIR', '/usr/lib')
+
+    conda_lib_dir = os.getenv('CONDA_LIB_DIR')
+    if conda_lib_dir is None:
+        conda_prefix = os.getenv('CONDA_PREFIX')
+        if conda_prefix is None:
+            raise RuntimeError(
+                'Must set CONDA_LIB_DIR to conda environment lib directory'
+            )
+        conda_lib_dir = os.path.join(conda_prefix, 'lib')
+
+    Path(lib_dir).mkdir(parents=True, exist_ok=True)
+
+    # Set RUNPATH to conda lib directory to determine locations of
+    # conda-provided dependencies
```

Comment:
curious as to why this is needed?

Reply:
Trying to set up the `ldd` call in a way that reflects the actual libraries we link against in the build environment is otherwise a bit annoying. I couldn't make it work without using `shell=True` on the run call and invoking `ldd` with `conda run` from within a Python script that is already being run from within a conda environment, which seems both more convoluted and more brittle than explicitly mirroring the `RUNPATH` manipulation that is used in the final installation anyway.